src/main/kotlin/minesweeper/MineSweeperGameMain.kt

```diff
@@ -0,0 +1,46 @@
+package minesweeper
+
+import minesweeper.domain.GameStateNotify
+import minesweeper.domain.GameStatus
+import minesweeper.domain.MineGenerator
+import minesweeper.domain.MinePosition
+import minesweeper.domain.MinePositionGenerator
+import minesweeper.domain.MinesMap
+import minesweeper.domain.MinesWeeperGame
+import minesweeper.domain.NumberValue
+import minesweeper.view.InputType
+import minesweeper.view.InputView
+import minesweeper.view.OutputView
+
+fun main() {
+
+    val height = NumberValue.of(InputView.inputDataFromConsole(InputType.HEIGHT))
+    val width = NumberValue.of(InputView.inputDataFromConsole(InputType.WIDTH))
+    val mineCount = NumberValue.of(InputView.inputDataFromConsole(InputType.MINE_COUNT))
+
+    val minesWeeperGame = MinesWeeperGame(width to height, mineCount)
+    minesWeeperGame.generateMine(object : MinePositionGenerator {
+        override fun generatePosition(): MinePosition {
+            return MineGenerator.generate(height, width)
+        }
+    })
```

Comment:
이 부분은 SAM Conversion을 적용하면 아래처럼 깔끔하게 lambda로 작성할 수 있습니다.

```suggestion
    gameMap.generateMine { minePosition }
```

Reply:
https://kotlinlang.org/docs/fun-interfaces.html

위 문서를 읽어보시면 될 것 같아요. 일단 핵심은 기존 인터페이스가 아래처럼 fun interface로 정의되어야 SAM Conversion을 사용할 수 있다는 점입니다. 이 부분은 코틀린의 버전이나 세부적인 부분에 따라 변경되었거나 변경될 수 있으니 참고 부탁드려요.

```kotlin
fun interface MinePositionGenerator {
    fun generatePosition(): MinePosition
}
```

Reply:
한수 배웠습니다. 감사합니다!